### DISCUSSION TITLE
Does i18n.changeLanguage not work any more?

### DISCUSSION THREAD

**Discussion by techgeniuswang** [i18next/next-i18next#1654] [Q&A] [answered]
I maintained a legacy application which used next-i18next (version **0.30.0**, ).
next-i18next : **0.30.0**
next.js: 7.0.3,
react: 16.14.0
In this case the next-i18next can use i18n.changeLanguage to load different language JSON file without change browser URL.

And we plan upgrade react/next.js/next-i18next to below version
next-i18next : **10.2.0**
next.js: 11.1.4
react: 17.0.2

I create a fresh application by [create-next-app](https://nextjs.org/learn/basics/create-nextjs-app/setup) then add next-i18next file. I found:
1) the latest version next-i18next is use **locale** to change language, and the locale append at last of URL like http://localhost:3000/zh
2) also if current locale equal to default locale, the "i18n.changeLanguage" can not work(http://localhost:3000/zh ). It can work only if current locale not equal to default locale (http://localhost:3000,)

here is a demo: https://3000-techgeniuswang-nextjsblo-k98dri25xg9.ws-eu29.gitpod.io/

first line button code is
`          <Link
            href='/'
            locale={router.locale === 'en' ? 'zh' : 'en'}
          >
            <button>
              {t('change-locale')}
            </button>
          </Link>  <span> change router.locale, </span>` 

second line button code is:
`        <button
          type='button'
          onClick={() => i18n.changeLanguage(i18n.language === 'en' ? 'zh' : 'en')}
        >
          {t('change-locale')}
        </button>  <span> change i18n.language</span>`

my question is:
1) does next-i18next support put locale in the middle of URL? like this: http://localhost:3000/path1/path2/zh/article1, http://localhost:3000/path1/path2/japanese/article2
2) because we expect the URL would not change when switch language, so is there a way use i18n.changeLanguage when locale equal to default locale

**Comment by isaachinman** [comment, marked as answer]
[NextJs handles locales and URLs now](https://nextjs.org/docs/advanced-features/i18n-routing), out of the box. You should no longer use `i18n.changeLanguage`, as it will have no effect on your NextJs app.

Your specific questions regarding URLs should be directed to the NextJs repo, as `next-i18next` no longer controls that functionality.

**Reply by techgeniuswang** [reply]
OK, Thank you for your clarification.
